spec/kicker.spec.js

```diff
@@ -221,9 +225,62 @@ describe('Kicker', function () {
             writerMock.verify()
           })
       })
+
+      after(function () {
+        processHostMock.restore()
+        writerMock.restore()
+      })
+    })
+
+    describe('when timeout is not set', function () {
+      before(function () {
+        log.reset()
+        delete kicker.configuration.changeWait
+        writerMock.restore()
+        processHostMock.restore()
+        writerMock = sinon.mock(writer)
+        processHostMock = sinon.mock(processHost)
+        processHostMock
+          .expects('restart')
+          .withArgs(kicker.configuration, kicker.writeFiles, kicker.onExit)
+          .callsArg(1)
+          .resolves({})
+        writerMock
+          .expects('hasFiles')
+          .withArgs(kicker.configuration)
+          .returns(true)
+        writerMock
+          .expects('writeFiles')
+          .withArgs(kicker.configuration)
+          .resolves(true)
+        kicker.timeout = setTimeout(() => {
+          log.info('We waited a long time for this!')
+        }, 1000000)
+      })
+
+      it('should use the default timeout', function (done) {
+        const mochaTestTimeoutForThisTest = 1000
+        this.timeout(mochaTestTimeoutForThisTest)
+
+        setTimeout(done, mochaTestTimeoutForThisTest - 100)
+        kicker.wait({ node: { key: 'nada' } })
+          .then(() => {
+            log.entries.should.eql([
```

Comment:
Maybe I'm confused by this test but wouldn't this *not* fail if the callback never gets called? In line 265, we set a timer to call `done` which happens no matter if this callback gets triggered or not, or am I missing something? (Very well possible! 🙂)

Reply:
You are right - good catch! Trying to work out the best way to fix this :)

Reply:
Alright, so I can bypass the 10 second timeout that's used by default by pre-resolving the promise from `wait` before returning it to mocha. That way I don't have to monkey about with `done` and it fails if the wrong timeout gets reported.